llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -2648,3 +2688,112 @@ Expected<DIEDataRef> DIEDataRef::create(MCCASBuilder &MB,
   append_range(B->Data, toStringRef(Contents));
   return get(B->build());
 }
+
+// Returns true if DIE should be placed in a separate CAS block.
+static bool shouldCreateSeparateBlockFor(DWARFDie &DIE) {
+  dwarf::Tag Tag = DIE.getTag();
+  if (!is_contained({dwarf::Tag::DW_TAG_subprogram}, Tag))
```

Comment:
To future-proof this, why don't we make this function a member of `DIEToCASConverter` and add all the tags we care about into a denseset?

Reply:
It needs no private members of `DIEToCASConverter`, as such it doesn't need to be a member.

>  add all the tags we care about into a denseset?

The problem with this is that the queries aren't as simple as "is this tag in a blessed list?".
More likely, each tag will have its own heuristic, like the low_pc one below

Reply:
That is fair